### ISSUE TITLE
setup port forwarding

### ISSUE THREAD

**Issue by systemsiteseason** [michaelkocian/War3Connect3#1]
I have to forwarding port both the server and the client right?

**Comment by michaelkocian** [comment]
No, you need to forward the port on the server and run this app on the client. Currently, only the warcraft v1.27.1.7085 version is supported. Tell me your version and I can tell you how to make it work.

**Comment by systemsiteseason** [comment]
I use path 1.24e and 1.26a

**Comment by systemsiteseason** [comment]
![image](https://user-images.githubusercontent.com/30430562/104657890-5e5b9000-56f4-11eb-85e6-74cb8e673bae.png)

**Comment by michaelkocian** [comment]
For the different versions, I can make it work.
But the person who would host the games and have port 6112 redirected on his router to the IP of his PC,
make sure he has a public IP reachable from the internet too. Not every internet provider gives public IP addresses.

**Comment by systemsiteseason** [comment]
i use vps and setting port forwarding 6112 both udp and tcp inbound and outbound

**Comment by systemsiteseason** [comment]
ok. i made it work. Can you give me an idea for it to work with the host bot? (ghost ++)

**Comment by michaelkocian** [comment]
I am not familiar with the host bot but I can tell you how this works.

1. When you press Lan games the warcraft broadcast a UDP packet in which it asks for all hosted games in the local network.
2. If this packet is caught by another PC that has a warcraft 3 hosted game it sends a response to the caller's IP with the information about the hosted game.
3. My application can send a packet like this to a certain IP address over the internet and if it reaches the PC mentioned in step 2 the response is sent over the internet too and it appears as a Lan game.
4. Once the warcraft on the client's side shows a hosted game it knows the IP and when you click, it initiates a direct TCP connection and if the port is redirected to the PC running warcraft it again connects.

Therefore I was able to host a warcraft 3 Lan game with the port 6112 redirected on my router to my PC. And everyone who ran this app saw me in Lan games. 

The last thing is that every warcraft 3 version sends a slightly different UDP packet containing the version thus it is easy to list all the versions and allow a selection if it would be needed.

**Comment by systemsiteseason** [comment]
Do you have any way to send to free port? because it uses SendTO

**Comment by michaelkocian** [comment]
What do you mean by a free port?
